### PR TITLE
Allow FreeDraw shapes to end close to their start

### LINKED ISSUE #84: Freedraw disappears when ending in the same location

### PR DESCRIPTION
**Pull request by alanhogan**
✓ New initially-failing tests
✓ Aaaand they pass
✓ Aaaaand one of them showed me that I was forgetting to include the ending position in the distance calculation so that's fixed

Was a pretty nice experience overall.

Fixes #84.

_Before_
goat 84 fail

_After_
goat 84 win

### CODE REVIEW

File: src/Goat/Update.elm [hunk 1]
```diff
@@ -369,6 +369,25 @@ finishValidDrawing model drawingInfo =
...
+    distance
+        + (calcDistance position previousPosition)
+        => position
+
+
+finishFreeHandDrawing : EditState -> DrawingInfo -> Model -> Model
+finishFreeHandDrawing finishedEditState ({ start, curPos, positions } as drawingInfo) model =
+    let
+        ( distance, _ ) =
+            List.foldl foldDistance ( 0.0, start ) (positions ++ [ curPos ])
```
Comment (alanhogan) [inline src/Goat/Update.elm:383]: error message was nice when I tried to do this:

```elm
(positions ++ curPos)
```

File: src/Goat/Update.elm [hunk 2]
```diff
@@ -369,6 +369,25 @@ finishValidDrawing model drawingInfo =
             model
 
 
+foldDistance : Position -> ( Float, Position ) -> ( Float, Position )
+foldDistance position ( distance, previousPosition ) =
```
Comment (gregziegan) [inline src/Goat/Update.elm:373]: `fold` is the higher order function, like `map` or `reduce`. So maybe this should be called sumDistances?

File: src/Goat/Update.elm (hunk 1 above)
Comment (gregziegan) [reply src/Goat/Update.elm:383]: would be interesting to [benchmark](http://package.elm-lang.org/packages/BrianHicks/elm-benchmark/latest) whether it's faster to do what you did or:
```elm
List.foldr foldDistance ( 0.0, start ) (curPos :: (List.reverse positions))
```

File: tests/src/Update.elm [hunk 3]
```diff
@@ -110,6 +110,37 @@ all =
                     |> .present
                     |> Array.isEmpty
                     |> Expect.true "drawing should not be added to array"
+        , test "should cancel drawing for tiny free-draw drawing" <|
+            \() ->
+                model
+                    |> update (ChangeDrawing DrawFreeHand)
+                    |> update (StartDrawing start)
+                    |> update (FinishDrawing (shiftPosition 2 2 start))
+                    |> .edits
```
Comment (gregziegan) [inline tests/src/Update.elm:119]: I wrote a little helper function `getFirstAnnotation`. You could put that here and then do `Expect.equal Nothing`. But I don't have a preference :D

**Review by gregziegan — APPROVED** [review]
all minor code style comments

**Comment by gregziegan** [comment]
merged bc it's a good fix for the next release. can (maybe?) rename things whenever

**Pull request merged**